Assets/Scripts/PlayerHands.cs

```diff
@@ -5,16 +5,119 @@
 
 public class PlayerHands : MonoBehaviour
 {
-    [SerializeField] private GameObject Torch;
-    [SerializeField] private bool isHoldingTorch = false;
+    [SerializeField] private GameObject playerTorch;
+    [SerializeField] private GameObject playerLighter;
+    [SerializeField] private GameObject torchPrefab;
+    [SerializeField] private Transform playerView;
+    [SerializeField] private Light playerLight;
+    [SerializeField] private bool isPlayerLightOn;
+    [SerializeField] private float torchLightRange = 8f;
+    [SerializeField] private float lighterLightRange = 3f;
+    [SerializeField] private float torchLightIntensity = 5.5f;
+    [SerializeField] private float lighterLightIntensity = 1f;
+    [SerializeField] private float throwForce = 1200f;
+    [SerializeField] private int lighterChance = 5;
```

Comment:
I think you see that there is too many things, and it becomes hard to maintain.
Class should be responsible for one thing. There is at least 3, keeping objects, lighter logic, light.
This class should be splited to smaller classes or (i recommend this option) some of its responsorialnie should be move to items (ligter).
Player is one of the most complicated systems to make, so i will leave it as it is.